query/src/main/java/org/infinispan/query/clustered/commandworkers/CQDelete.java

```diff
@@ -27,16 +26,14 @@ CompletionStage<QueryResponse> perform(BitSet segments) {
       }
 
       SearchQueryBuilder query = queryDefinition.getSearchQueryBuilder();
-      return blockingManager.supplyBlocking(() -> fetchIds(query), this)
-            .thenApply(queryResult -> queryResult.hits().stream().map(id -> cache.remove(id) != null ? 1 : 0).reduce(0, Integer::sum))
```

Comment:
Can we not just use `cache.removeAsync` instead and use `thenCompose` instead of `thenApply`?